packages/core/src/core/coreToolScheduler.test.ts

```
@@ -30,116 +29,91 @@ import {
+    matchingCall = latestCalls?.find((call) => call.status === status);
+    if (!matchingCall) {
+      throw new Error(
+        `Waiting for status "${status}", latest statuses: ${
+          latestCalls?.map((call) => call.status).join(', ') ?? 'none'
+        }`,
+      );
+    }
   });
+  return matchingCall;
 }
+import type { ContextAwareTool, ToolContext } from '../tools/tool-context.js';
```

Biome (2.1.2)

[error] Shouldn't redeclare 'ContextAwareTool'. Consider to delete it or rename it.

'ContextAwareTool' is defined here:


(lint/suspicious/noRedeclare)